geomesa-native-api/src/main/java/org/locationtech/geomesa/api/GeoMesaQuery.java

```diff
@@ -45,11 +46,62 @@ public GeoMesaQueryBuilder within(double lx, double ly, double ux, double uy) {
         }
 
         public GeoMesaQueryBuilder during(Date start, Date end) {
-            this.start = start;
-            this.end = end;
+            return during(start, true, end, true);
+        }
+
+        /**
+         * We implement the "during" as a GeoTools "between" predicate, which includes the endpoints
+         * by default; see:
+         *   https://github.com/geotools/geotools/blob/15.x/modules/library/main/src/main/java/org/geotools/filter/IsBetweenImpl.java#L144
+         *
+         * If the temporal bounds are set to non-sensical values, that will be handled elsewhere.
+         *
+         * @param start the start date
+         * @param isStartInclusive whether the start date should be included in the interval
+         * @param end the end date
+         * @param isEndInclusive whether the end date should be included in the interval
+         * @return the query object updated for the effective temporal bounds
+         */
+        protected GeoMesaQueryBuilder during(Date start, boolean isStartInclusive, Date end, boolean isEndInclusive) {
+            if (isStartInclusive) {
+                this.start = start;
+            } else {
+                this.start = start != null ? new Date(start.getTime() + 1L) : null;
+            }
+
+            if (isEndInclusive) {
+                this.end = end;
+            } else {
+                this.end = end != null ? new Date(end.getTime() - 1L) : null;
+            }
+
             return this;
         }
 
+        /**
+         * Converts the current start- and end-point into a valid temporal filter.
+         *
+         * @return the net temporal filter that results from the current temporal bounds
+         */
+        public Filter getTemporalFilter() {
```

Comment:
I like this implementation.

Given it, it looks like we could offer 'after' and 'before', no?

Also, those could be combined.  Builder.after(start).before(end).build would return a between.